scripts/fetchai_netutils/fetch/testing/testcase.py

```diff
@@ -0,0 +1,477 @@
+            files = os.listdir(load_from)
+
+            for f in files:
+                shutil.copy(load_from + f, root)
+
+        port = self._port_start_range + (self._port_range * index)
+        key = self._nodes_keys[index]
+        pub = self._nodes_pubkeys[index]
+
+        # Create an instance of the constellation - note we don't clear path since
+        # it should be clear unless load_directory is used
+        instance = DmlfEtchInstance(
+            self._node_exe,
+            port,
+            key,
+            pub,
+            root,
+            clear_path=False
+        )
+
+        assert len(self._nodes) == index, "Attempt to add node with an index mismatch. Current len: {}, index: {}".format(
+            len(self._nodes), index)
+
+        self._nodes.append(instance)
+
+    def setup_nodes_keys(self):
+
+        # Give each node a unique identity
+        for index in range(self._number_of_nodes):
+            entity = Entity()
+            pub64 = entity.public_key
+            key64 = entity.private_key
+
+            print('Setting up key for node {}...'.format(index))
+            print('Giving node the identity: {}'.format(pub64))
+            self._nodes_pubkeys.append(pub64)
+
+            print('Giving node key: {}'.format(key64))
+            self._nodes_keys.append(key64)
+
+    def connect_nodes(self, node_connections):
+        for connect_from, connect_to in node_connections:
+            self._nodes[connect_from].add_peer(self._nodes[connect_to])
+            output("Connect node {} to {}".format(connect_from, connect_to))
+
+    def start_node(self, index):
+        print('Starting Node {}...'.format(index))
+
+        self._nodes[index].start()
+        print('Starting Node {}...complete'.format(index))
+
+        time.sleep(1)
+
+    def restart_node(self, index):
+        print('Restarting Node {}...'.format(index))
+
+        self._nodes[index].stop()
+
+        # Optically remove db files when testing recovering from a genesis file
+        if False:
+            self.dump_debug(index)
+
+            pattern = ["*.db"]
+            for p in pattern:
+                [os.remove(x) for x in glob.iglob('./**/' + p, recursive=True)]
+
+        self.start_node(index)
+        time.sleep(3)
+
+    def print_time_elapsed(self):
+        output("Elapsed time: {}".format(
+            time.perf_counter() - self._creation_time))
+
+    def run(self):
+
+        # setup node keys
+        self.setup_nodes_keys()
+
+        # build up all the node instances
+        for index in range(self._number_of_nodes):
+            self.append_node(index, self._node_load_directory)
+
+        # Now connect the nodes as specified
+        if self._node_connections:
+            self.connect_nodes(self._node_connections)
+
+        # start all the nodes
+        for index in range(self._number_of_nodes):
+            self.start_node(index)
+
+        time.sleep(5)  # TODO(HUT): blocking http call to node for ready state
```

Comment:
You might not need this sleep for your use case

Reply:
yes, I can sleep as a step